### PR TITLE
feat(argo-cd): add sync wave delay option for application controller

### PR DESCRIPTION
**Pull request by ssbostan**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by ssbostan** [comment]
This commit is related to allowing the user to override the default ARGOCD_SYNC_WAVE_DELAY env. variable.

File: charts/argo-cd/values.yaml [hunk 1]
```diff
@@ -248,6 +248,8 @@ configs:
     controller.self.heal.timeout.seconds: 5
     # -- Repo server RPC call timeout seconds.
     controller.repo.server.timeout.seconds: 60
+    # -- Sync wave delay seconds to make delay between applications' sync
+    controller.sync.wave.delay: 2
```
Comment (yu-croco) [inline charts/argo-cd/values.yaml:252]: 📝 default is `2` 
https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/
image

**Comment by yu-croco** [comment]
Hi @ssbostan , thank you for your PR, it's almost LGTM.
In order to pass lint, you update README.md.gotmpl and run `./scripts/helm-docs.sh`, instead of updating README directly.
Ref: https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation

**Comment by mkilchhofer** [comment]
Hi @ssbostan
Thanks for the contribution.
Is there a reference in upstream manifests of this env variable? I dind't find it there.

Would you mind to also file a PR against upstream and contribute the new setting in the params ConfigMap also there?

**Comment by ssbostan** [comment]
> Hi @ssbostan Thanks for the contribution. Is there a reference in upstream manifests of this env variable? I dind't find it there.
> 
> Would you mind to also file a PR against upstream and contribute the new setting in the params ConfigMap also there?

Yes, please. Could you send me the related information to commit on the upstream as well?

Thank you.

**Pull request closed without merging**